auditlog/tests/test_auditlog.py

```diff
@@ -181,3 +182,101 @@ def test_03_copy_res_partner_logging_old_api(self):
         ])
 
         self.assertEqual(len(logs), 1)
+
+
+class TestFieldRemoval(TransactionCase):
+    def setUp(self):
+        super(TestFieldRemoval, self).setUp()
+
+        # Clear all existing logging lines
+        existing_audit_logs = self.env['auditlog.log'].search([])
+        existing_audit_logs.unlink()
+
+        # Create a test model to remove
+        self.test_model = self.env['ir.model'].create({
+            'name': 'x_test_model',
+            'model': "x_test.model",
+            'state': 'manual'
+        })
+
+        # Create a test model field to remove
+        self.test_field = self.env['ir.model.fields'].create({
+            'name': 'x_test_field',
+            'field_description': 'x_Test Field',
+            'model_id': self.test_model.id,
+            'ttype': 'char',
+            'state': 'manual'
+        })
+
+        # Setup auditlog rule
+        self.auditlog_rule = self.env['auditlog.rule'].create({
+            'name': 'test.model',
+            'model_id': self.test_model.id,
+            'log_type': 'fast',
+            'log_read': False,
+            'log_create': True,
+            'log_write': True,
+            'log_unlink': False,
+            'log_custom_method': True,
+        })
+
+        self.auditlog_rule.subscribe()
+
+    def tearDown(self):
+        """ Clean up after testcase. """
+        self.auditlog_rule.unsubscribe()
+        self.auditlog_rule.unlink()
+        self.registry.models.pop('x_test.model')
+        self.registry._pure_function_fields.pop('x_test.model')
+
+        # Force reset of environment and clear cursor actions
+        self.env.reset()
+        self.cr.rollback()
```

Comment:
Thy do you need to perform all these SQL actions? Is that not why this runs in a test case?

Reply:
I believe during the model creation https://github.com/OCA/OCB/blob/8.0/openerp/addons/base/ir/ir_model.py#L222 the database cursor is commited. During the testcase, I pass MODULE_UNINSTALL_FLAG to the unlink function to avoid a registry reload which in turn triggers a complete rerun of the previous testcases (including this one) in this module and resulting in a duplicate model name constraint error.
My assumption here was that I manually needed to rollback all other cursor actions before I deleted the ir_model record. If you don't delete the ir_model record, the ACL warning pops up because the registry reloads everything from scratch on other test modules.
The testcase works fine without the above two statement, but I think this cleans up the auditlog_log_lines? Am I wrong?

Reply:
No, I think you are correct in your analysis but your solution looks void to me. A rollback only rolls back to the last commit so I don't think it helps you to circumvent the commits in _auto_init.

Try this before you create the model: https://github.com/OCA/OpenUpgrade/blob/8.0/openerp/modules/loading.py#L140-L141, then reassign the original commit method after the model creation.

Reply:
Indeed, that is the correct fix to solve the problem. Thanks for the tip!